x/trigger/keeper/event_listener.go

```diff
@@ -62,3 +112,43 @@ func (k Keeper) IterateEventListeners(ctx sdk.Context, eventName string, handle
 	}
 	return nil
 }
+
+// HasEventListener checks if a specific event listener exists
+func (k Keeper) HasEventListener(ctx sdk.Context, eventName string, order uint64, triggerID uint64) (bool, error) {
+	eventHash, err := getEventNameHash(eventName)
+	if err != nil {
+		return false, err
+	}
+	key := collections.Join3(eventHash, order, triggerID)
+	return k.EventListeners.Has(ctx, key)
+}
+
+// RemoveAllEventListenersForTrigger removes all event listeners for a specific trigger
+func (k Keeper) RemoveAllEventListenersForTrigger(ctx sdk.Context, triggerID uint64) error {
+	trigger, err := k.GetTrigger(ctx, triggerID)
+	if err != nil {
+		return err
+	}
+	_, err = k.RemoveEventListener(ctx, trigger)
+	return err
+}
```

Comment:
I'm confused by this. It only removes one event listener. From what I can tell, all it does is delete a trigger event based on a trigger id. I suggest renaming it to `RemoveEventListenerForTriggerID`.

I also don't think it's in use anywhere, so consider just deleting it.